Xamarin.Forms.Build.Tasks/XmlnsCache.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Mono.Cecil;
+using IOPath = System.IO.Path;
+
+namespace Xamarin.Forms.Build.Tasks
+{
+	class XmlnsCache : IDisposable
+	{
+		readonly string[] _references;
+		readonly List<XmlnsDefinitionAttribute> _xmlnsDefinitions = new List<XmlnsDefinitionAttribute>();
+		readonly Dictionary<string, ModuleDefinition> _xmlnsModules = new Dictionary<string, ModuleDefinition>();
+		bool _initialized;
+
+		public List<XmlnsDefinitionAttribute> XmlnsDefinitions
+		{
+			get { Initialize(); return _xmlnsDefinitions; }
+		}
+
+		public Dictionary<string, ModuleDefinition> XmlnsModules
+		{
+			get { Initialize(); return _xmlnsModules; }
+		}
+
+		public XmlnsCache(string[] references)
+		{
+			_references = references;
+		}
+
+		void Initialize()
+		{
+			if (_initialized)
+				return;
+			_initialized = true;
+
+			var paths = _references?.ToList() ?? new List<string>();
+			//Load xmlnsdef from Core and Xaml
+			paths.Add(typeof(Label).Assembly.Location);
+			paths.Add(typeof(Xamarin.Forms.Xaml.Extensions).Assembly.Location);
+
+			foreach (var path in paths.Distinct())
+			{
+				string asmName = IOPath.GetFileName(path);
+				if (AssemblyIsSystem(asmName))
+					// Skip the myriad "System." assemblies and others
+					continue;
+
+				using (var asmDef = AssemblyDefinition.ReadAssembly(path))
+				{
+					foreach (var ca in asmDef.CustomAttributes)
+					{
+						if (ca.AttributeType.FullName == typeof(XmlnsDefinitionAttribute).FullName)
+						{
+							_xmlnsDefinitions.Add(ca.GetXmlnsDefinition(asmDef));
+							_xmlnsModules[asmDef.FullName] = asmDef.MainModule;
+						}
+					}
+				}
+			}
+		}
+
+		bool AssemblyIsSystem(string name)
+		{
+			if (name.StartsWith("System.Maui", StringComparison.OrdinalIgnoreCase))
```

Comment:
We need to update this to be "Microsoft.Maui"

Reply:
@jsuarezruiz this PR can't apply to MAUI because XamlG is a Roslyn source generator now.

Since Xamarin.Forms 5.0 is basically service mode now, I'm inclined to just close this. What do you think?